### ISSUE TITLE
consider consolidating

### ISSUE THREAD

**Issue by tyrasd** [ideditor/schema-builder#35]
Let's discuss the suggestion from https://github.com/ideditor/schema-builder/commit/d9c23fac2a3d1733e55c841c1e9e476ad78e2f7f#commitcomment-64100888:

> [This repository could just be] some build scripts in the id-tagging-schema repo.

Right out of the box, I would agree.
I thought that there was a perhaps a reason for the repositories being structured like this. Perhaps some forks are using it to maintain their own set of presets?! Maybe @quincylvania could explain the original reasoning?

**Comment by 1ec5** [comment]
The idea probably was in pursuit of a schema-agnostic iD. I don’t know if any forks are maintaining their own tagging schemas, but in principle a tasking manager could be set up to tailor the presets to a task-specific set or something. JOSM has this capability with swappable preset packs, but I’m not sure about the level of demand for iD to support something similar.

**Comment by westnordost** [comment]
(FWIW as an irregular contributor, I (also) found it confusing which to find in in which repo)

**Comment by matkoniecz** [comment]
In other words: I support merging schema builder and iD tagging schema into one repo, but what kind of negatives it would bring us and to others?

How we can avoid this negatives?

Maybe we can put announcement of this plan into changelog, so people with potential interest can comment on? Before we potentially break something?

Or maybe there is noone who relies on this two-repo split and it will bring no negative consequences to the current users?

**Comment by k-yle** [comment]
next steps:
1. (blocker) release schema v7
2. do not accept any more PRs in this repository
3. update [#2431](https://github.com/openstreetmap/id-tagging-schema/pull/2431) to include the final commits from this repo
4. merge [#2431](https://github.com/openstreetmap/id-tagging-schema/pull/2431)

follow up work:

5. abandon all PRs in this repo, and re-open those PRs in the _id-tagging-schema_ repo. there are only 3 non-docs PRs, so this is fine.
6. batch-move all open issues to _id-tagging-schema_, and batch-label them as https://github.com/ideditor/schema-builder/labels/json-schema. there's an [AI Skill or CLI command](https://cli.github.com/manual/gh_issue_transfer) which can do this.
7. [deprecate the npm package](https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions)
8. [archive this repo on github](https://docs.github.com/en/repositories/archiving-a-github-repository/archiving-repositories)? [^1]
[^1]: not sure if this step is a good idea. it will prevent people from commenting or reacting to old closed PRs. maybe that's fine?

**Comment by matkoniecz** [comment]
> batch-move all open issues to id-tagging-schema, and batch-label them as json-schema . there's an [AI Skill or CLI command](https://cli.github.com/manual/gh_issue_transfer) which can do this.

can we label them here, and create matching label at id-tagging-schema? Or is moving issues removing labels anyway?

**Comment by k-yle** [comment]
> can we label them here, and create matching label at id-tagging-schema? Or is moving issues removing labels anyway?

yes, if we use the same label name, then it can be added before or after transfering the issue<sup>[[1]](https://docs.github.com/en/issues/tracking-your-work-with-issues/administering-issues/transferring-an-issue-to-another-repository#:~:text=Labels%20and%20milestones%20are%20also%20retained%20if%20they%27re%20present%20in%20the%20target%20repository%2C%20with%20labels%20matching%20by%20name%20and%20milestones%20matching%20by%20both%20name%20and%20due%20date.)</sup>

**Comment by k-yle** [comment]
unfortunately transferring issues between two different organisations involves [a rather amusing process](https://github.com/orgs/community/discussions/5766#discussioncomment-8699777), and requires more permissions than i have.

however, it does work. tested with https://github.com/osm-nz/osm-nz.github.io/issues/4


EDIT: actually, it partially succeeds. even though the graphql API returns an error. That's why https://github.com/openstreetmap/id-tagging-schema/issues/2434 got created, and now duplicates https://github.com/ideditor/schema-builder/issues/244

**Comment by tyrasd** [comment]
> the timestamp and author is wrong

isn't it that we want the ticket to have the original timestamp/author? If we correctly link them and close the issues in the old repo it should be fine, shouldn't it?

**Comment by matkoniecz** [comment]
> the timestamp and author is wrong

what you mean by it being wrong?

**Comment by matkoniecz** [comment]
I opened https://github.com/ideditor/schema-builder/pull/341 that would archive this repo by deleting most files and leaving some as a redirect to new home.

I think it makes more sense that keeping more and more diverging docs.

**Comment by k-yle** [comment]
> > the timestamp and author is wrong

> isn't it that we want the ticket to have the original timestamp/author? If we correctly link them and close the issues in the old repo it should be fine, shouldn't it?

yes correct, sorry for the poorly phrased comemnt.

but the fact that it works is a github bug. it also only transfers the issue, not the comments, and doesn't create a redirect. so [this process](https://github.com/orgs/community/discussions/5766#discussioncomment-8699777) is the only way to do it. 

that would require an admin of the [@ideditor](https://github.com/ideditor) org to create a new repo. i'm happy to help out with the rest of the process, but i can't do that part

**Comment by k-yle** [comment]
for future discussion on this topic, lets use https://github.com/openstreetmap/id-tagging-schema/issues/2449
